addons/mail/models/mail_message.py

```diff
@@ -988,8 +989,11 @@ def _to_store(
         for record in records:
             thread_data = {}
             if record._name != "discuss.channel":
-                # sudo: mail.thread - if mentionned in a non accessible thread, name is allowed
-                thread_data["name"] = record.sudo().display_name
+                try:
+                    # sudo: mail.thread - if mentionned in a non accessible thread, name is allowed
+                    thread_data["name"] = record.sudo().display_name
+                except MissingError:
+                    continue  # related non mail.thread document deleted, still show message in history
```

Comment:
I don't really like this try/except, shouldn't the fix be to add an `exists()` in `_records_by_model_name` after `self.env[message.model].browse(message.res_id)`